packages/mobile/src/components/audio/GoogleCast.tsx

```diff
@@ -130,24 +142,51 @@ export const useChromecast = () => {
     }
   }, [playing, playCast, pauseCast, castState])
 
+  // Set buffering state when cast is buffering
+  useEffect(() => {
+    if (
+      castState === CastState.CONNECTING ||
+      ((mediaStatus === undefined ||
+        mediaStatus?.playerState === undefined ||
+        mediaStatus?.playerState === MediaPlayerState.IDLE ||
+        mediaStatus?.playerState === MediaPlayerState.LOADING ||
+        mediaStatus?.playerState === MediaPlayerState.BUFFERING) &&
+        castState !== CastState.NOT_CONNECTED)
+    ) {
+      dispatch(playerActions.setBuffering({ buffering: true }))
+    } else {
+      dispatch(playerActions.setBuffering({ buffering: false }))
+    }
+  }, [mediaStatus, castState, dispatch])
+
   // Seek the cast device
   useEffect(() => {
     if (seek !== null) {
       client?.seek({ position: seek })
     }
   }, [client, seek])
 
+  // Mute the track player if we are connecting to cast
   useEffect(() => {
     if (
       castState === CastState.CONNECTED ||
```

Comment:
interesting so technically they are both playing, just one is muted?

Reply:
yeah. chromecast makes no assumptions about what's going on on the device, but since we want the progress bar to sync up, etc., found this the easiest way when we implemented chromecast way back when (before reloaded even). down to revisit.